src/ert/_c_wrappers/enkf/res_config.py

```diff
@@ -405,11 +420,88 @@ def _alloc_from_dict(self, config_dict):
 
         self.model_config = ModelConfig(
             data_root=self.config_path,
-            joblist=self.site_config.job_list,
+            joblist=self.job_list,
             refcase=self.ensemble_config.refcase,
             config_dict=config_dict,
         )
 
+    def _read_env_vars(self, user_config_dict, site_config_dict):
+        env_vars = EnvironmentVarlist()
+
+        def add_from_dict(config_dict):
+            environment_vars = config_dict.get(ConfigKeys.SETENV, [])
+
+            for item in iter(environment_vars):
+                if isinstance(item, Mapping):
+                    key = item["NAME"]
+                    value = item["VALUE"]
+                else:
+                    key, value = item
+                env_vars.setenv(key, value)
+
+            paths = config_dict.get("UPDATE_PATH", [])
+
+            for key, value in iter(paths):
+                env_vars.update_path(key, value)
+
+        add_from_dict(user_config_dict)
+        add_from_dict(site_config_dict)
+        return env_vars
+
+    def _read_install_job(self, licence_root_path, user_config_dict, site_config_dict):
+        jobs = ExtJoblist()
+
+        def add_from_dict(config_dict):
+
+            for job in config_dict.get(ConfigKeys.INSTALL_JOB, []):
+                if isinstance(job, Mapping):
+                    path = job[ConfigKeys.PATH]
+                    name = job[ConfigKeys.NAME]
+                else:
+                    name, path = job
+                extjob = ExtJob(
+                    license_root_path=licence_root_path,
+                    private=False,
+                    config_file=os.path.abspath(path),
+                    name=name,
+                )
+                jobs.add_job(extjob.name(), extjob)
+
+            if ConfigKeys.INSTALL_JOB_DIRECTORY in config_dict:
+                job_paths = config_dict[ConfigKeys.INSTALL_JOB_DIRECTORY]
+                if isinstance(job_paths, str):
+                    job_paths = [job_paths]
+                for job_path in job_paths:
+                    if not os.path.isdir(job_path):
+                        print(f"WARNING: Unable to locate job directory {job_path}")
+                        continue
+                    files = os.listdir(job_path)
+                    for file_name in files:
+                        full_path = os.path.abspath(os.path.join(job_path, file_name))
```

Comment:
... and then `full_path` can become `job_path`, if you feel thusly inclined

Reply:
I did something to some things around this thing.

Reply:
beautiful.